specification/batch/data-plane/Microsoft.Batch/stable/2021-01-01.13.0/BatchService.json

```diff
@@ -0,0 +1,17255 @@
+            },
+            "description": "A response containing the Task counts for the specified Job.",
+            "schema": {
+              "$ref": "#/definitions/TaskCountsResult"
+            }
+          },
+          "default": {
+            "description": "The error from the Batch service.",
+            "schema": {
+              "$ref": "#/definitions/BatchError"
+            }
+          }
+        }
+      }
+    },
+    "/pools": {
+      "post": {
+        "tags": [
+          "Pools"
+        ],
+        "operationId": "Pool_Add",
+        "x-ms-examples": {
+          "Add a CloudServiceConfiguration pool": {
+            "$ref": "./examples/PoolAdd_CloudServiceConfiguration.json"
+          },
+          "Add a VirtualMachineConfiguration pool": {
+            "$ref": "./examples/PoolAdd_VirtualMachineConfiguration.json"
+          },
+          "Add a VirtualMachineConfiguration pool with containers": {
+            "$ref": "./examples/PoolAdd_VirtualMachineConfigurationWithContainers.json"
+          },
+          "Add a VirtualMachineConfiguration pool with extensions": {
+            "$ref": "./examples/PoolAdd_VirtualMachineConfigurationWithExtensions.json"
+          },
+          "Add a pool with mount drive specified": {
+            "$ref": "./examples/PoolAdd_MountConfiguration.json"
+          }
+        },
+        "summary": "Adds a Pool to the specified Account.",
+        "description": "When naming Pools, avoid including sensitive information such as user names or secret project names. This information may appear in telemetry logs accessible to Microsoft Support engineers.",
+        "x-ms-request-id": "request-id",
+        "parameters": [
+          {
+            "name": "pool",
+            "in": "body",
+            "description": "The Pool to be added.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PoolAddParameter"
+            }
+          },
+          {
+            "name": "timeout",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "format": "int32",
+            "default": 30,
+            "description": "The maximum time that the server can spend processing the request, in seconds. The default is 30 seconds.",
+            "x-ms-parameter-grouping": {
+              "postfix": "Options"
+            }
+          },
+          {
+            "name": "client-request-id",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "format": "uuid",
+            "description": "The caller-generated request identity, in the form of a GUID with no decoration such as curly braces, e.g. 9C4D50EE-2D56-4CD3-8152-34347DC9F2B0.",
+            "x-ms-parameter-grouping": {
+              "postfix": "Options"
+            },
+            "x-ms-client-request-id": true
+          },
+          {
+            "name": "return-client-request-id",
+            "defau
```

Comment:
Are there properties in the NodeVMExtension object model (and the models it composes) that are readonly? if so consider marking them as readonly, which ensures unnecessary setters are not generated / better user experience.

Reply:
I went ahead and marked the instance view & status properties as read-only, but I'll leave this comment open for now since I'd like someone else from the service team to review these. (@xingwu1 when you're around mind taking a look?)